### ISSUE TITLE
qvm-shutdown --all should shutdown all DomUs in parallel

### ISSUE THREAD

**Issue by DemiMarie** [QubesOS/qubes-issues#4572]
### Qubes OS version:

`Qubes release 4.0 (R4.0)`

### Affected component(s): 

qubes-core-admin

---

### Steps to reproduce the behavior:

Run `qvm-shutdown --all`

### Expected behavior:

Qubes are all shutdown in parallel, as is done by `xl shutdown -a`.

### Actual behavior:

Qubes doesn’t shutdown NetVMs until all VMs that they are providing networks for are destroyed.

### General notes:

In general, shutting down a NetVM that provides networking for an AppVM is a bad thing, but if all VMs are being shutdown, then it is okay.

---

**Comment by marmarek** [comment]
You can achieve it with `qvm-shutdown --all --wait` - it will wait until all VMs are shutdown, including retrying shutdown when dependent VMs are already off.

**Comment by DemiMarie** [comment]
@marmarek That works, but has two problems:

* It sometimes fails with `Got an empty response from qubesd`.
* It is slow

My understanding is that when shutting down all VMs, there is no need to delay a VM’s shutdown until all of its dependent VMs are also shutdown.  Furthermore, this could happen anyway, since a VM can shut itself down at any time, so Qubes needs to handle that.

**Comment by andrewdavidwong** [comment]
> My understanding is that when shutting down all VMs, there is no need to delay a VM’s shutdown until all of its dependent VMs are also shutdown.

Reopening for this part, unless of course @marmarek says it's a `won't do`.

> Furthermore, a VM can shut itself down at any time, so Qubes needs to handle that.

Handle it how? I don't see what this has to do with the issue as described.

**Comment by DemiMarie** [comment]
@andrewdavidwong One could always run `shutdown -h now` in a VM, or the VM might kernel panic.  Qubes needs to handle this gracefully, without affecting the rest of the system excessively, to avoid denial-of-service vulnerabilities.  Qubes does do this, very well.

As such, refusing to shut down a VM when there are VMs that depend on it seems to be overly conservative.  Perhaps a `--force` option on the command line tool?
